weblate/glossary/forms.py

```diff
@@ -76,11 +52,10 @@ def __init__(self, unit, user, data=None, instance=None, initial=None, **kwargs)
             component__in=component.project.glossaries,
             component__manage_units=True,
         )
-        self._user = user
         exclude = [
             glossary.pk
             for glossary in glossaries
-            if not user.has_perm("unit.add", glossary)
+            if not user.has_perm("glossary.add", glossary)
```

Comment:
This is not necessary, as the permission check handles this automatically:

https://github.com/WeblateOrg/weblate/blob/613545fe502a5d350a8d174ef240b215173de99c/weblate/auth/permissions.py#L310-L325

```suggestion
            if not user.has_perm("unit.add", glossary)
```

Reply:
According to my testing, if I don’t modify this (keep `"unit.add"`), the “Add term to glossary” button will not be displayed in some cases (I guess that this issue exists in monolingual files, but I have not thoroughly tested it).

<details>
<summary>Screenshots</summary>

![image 1](https://github.com/WeblateOrg/weblate/assets/30619816/ec531fe3-dfef-4c5d-ae43-dc8cdd321be0)
![image 2](https://github.com/WeblateOrg/weblate/assets/30619816/5ecf8515-2bd7-4d45-b671-b54db77b1e19)
</details>

Reply:
Indeed the `unit.add` has some additional checks, but these are necessary to make it work (like checking whether used file format can actually add strings).

Reply:
The reason for the inconsistency between the two behaviours is due to this check:

https://github.com/WeblateOrg/weblate/blob/59d6a63cedb364fdb1a5177eb2f819883a737a28/weblate/auth/permissions.py#L263-L272

This check prevents adding strings to a monolingual file that are not in the source language. I think this check is unnecessary for a glossary, because sometimes users need to add terms in non-source languages. Anyway, I will revert this commit.